src/main/java/org/jabref/gui/mergeentries/MultiMergeEntriesViewModel.java

```diff
@@ -0,0 +1,112 @@
+package org.jabref.gui.mergeentries;
+
+import java.util.Map;
+import java.util.function.Supplier;
+
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.ListProperty;
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.property.SimpleListProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.property.StringProperty;
+import javafx.collections.FXCollections;
+import javafx.scene.control.ButtonType;
+
+import org.jabref.gui.AbstractViewModel;
+import org.jabref.gui.util.BackgroundTask;
+import org.jabref.gui.util.TaskExecutor;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+
+public class MultiMergeEntriesViewModel extends AbstractViewModel {
+
+    private final ListProperty<Entry> entries = new SimpleListProperty<>(FXCollections.observableArrayList());
+
+    private final ObjectProperty<BibEntry> mergedEntry = new SimpleObjectProperty<>(new BibEntry());
+
+    private final ListProperty<String> failedSuppliers = new SimpleListProperty<>(FXCollections.observableArrayList());
+
+    public void addSource(Entry entryColumn) {
+        if (!entryColumn.isLoading.getValue()) {
+            updateFields(entryColumn.entry.get());
+        } else {
+            entryColumn.isLoading.addListener((observable, oldValue, newValue) -> {
+                if (!newValue) {
+                    updateFields(entryColumn.entry.get());
+                    if (entryColumn.entryProperty().get() == null) {
+                        failedSuppliers.add(entryColumn.titleProperty().get());
+                    }
+                }
+            });
+        }
+        entries.add(entryColumn);
+    }
+
+    public void updateFields(BibEntry entry) {
+        if (entry == null) {
+            return;
+        }
+        for (Map.Entry<Field, String> fieldEntry : entry.getFieldMap().entrySet()) {
+            // make sure there is a row for the field
+            if (!mergedEntry.get().getFieldsObservable().containsKey(fieldEntry.getKey())) {
+                mergedEntry.get().setField(fieldEntry.getKey(), fieldEntry.getValue());
+            }
+        }
+    }
+
+    public BibEntry resultConverter(ButtonType button) {
+        if (button == ButtonType.CANCEL) {
+            return null;
+        }
+        return mergedEntry.get();
+    }
+
+    public ListProperty<Entry> entriesProperty() {
+        return entries;
+    }
+
+    public ObjectProperty<BibEntry> mergedEntryProperty() {
+        return mergedEntry;
+    }
+
+    public ListProperty<String> failedSuppliersProperty() {
+        return failedSuppliers;
+    }
+
+    public static class Entry {
```

Comment:
Maybe rename to MergeEntry, because otherwise it's confusing with BibEntry and Entry

Reply:
Makes sense, but I chose EntrySource instead because it is one option for the merge, not the merged entry.